src/main/java/net/pms/encoders/RAWThumbnailer.java

```diff
@@ -98,37 +103,67 @@ public int type() {
 		return Format.IMAGE;
 	}
 
-	public static byte[] getThumbnail(OutputParams params, String fileName) throws IOException {
+	public static byte[] getThumbnail(OutputParams params, String fileName, ImageInfo imageInfo) throws IOException {
 		// Use device-specific pms conf
 		PmsConfiguration configuration = PMS.getConfiguration(params);
 		params.log = false;
 
-		String cmdArray[] = new String[4];
+		// This is a wild guess at a decent buffer size for an embedded thumbnail.
+		// Every time the buffer has to grow, the whole buffer must be copied in memory.
+		params.outputByteArrayStreamBufferSize = 300000;
+
+		// First try to get the embedded thumbnail
+		String cmdArray[] = new String[6];
 		cmdArray[0] = configuration.getDCRawPath();
 		cmdArray[1] = "-e";
 		cmdArray[2] = "-c";
-		cmdArray[3] = fileName;
-		ProcessWrapperImpl pw = new ProcessWrapperImpl(cmdArray, params);
+		cmdArray[3] = "-t";
+		cmdArray[4] = Integer.toString(imageInfo != null ? exifOrientationToDCRaw(imageInfo.getExifOrientation()) : 0);
```

Comment:
@Nadahar Why did you set this array ? As i understood it, by default DCRaw will use the orientation "-t" set in the EXIF metadata.

As it is now, it's not working. You can test it in your Web interface.